platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -15,4 +15,34 @@
 [[best_practices_general]]
 = General Gradle Best Practices
 
-Add content here.
+== Do Not Use Internal APIs
+
+Do not use APIs from a package where any segment of the package is `internal`, or types that have `Internal` or `Impl` as a suffix in the name.
```

Comment:
It might be a good idea to point to everything that the Gradle team thinks is public from here? I know we have https://github.com/gradle/gradle/blob/9260f503383753c0de270f3d1843b8ad328825dd/build-logic-commons/basics/src/main/kotlin/gradlebuild/basics/PublicApi.kt#L12-L50, though I am not sure there is some user facing documentation about this.

Reply:
I'm happy to add that a follow once once such page exists, unless you are saying i should that copy that list here.

Reply:
I suppose I'd like to get some input from @lkasso here...

Reply:
> Linking to a published public API artifact containing only the public API classes might be more useful.

I was under the impression that Gradle has no such artifact. AGP and KGP does have those so I could link to those. WDYT?

Reply:
It will be documented in: https://github.com/gradle/gradle/pull/32902. @liutikas, feel free to point to it.

Reply:
Let's do that as a follow up.